test/libsolidity/ASTPropertyTests/nested_properties.sol

```diff
@@ -0,0 +1,17 @@
+contract C {
+    function f() public pure {
+        /// TestCase1: condition.operator
+        for(uint i = 0; i < 42; ++i) {
+        }
+        /// TestCase2: initializationExpression.initialValue.value
+        for(uint i = 1; i < 42; i = i * 2) {
+        }
+        /// TestCase3: loopExpression.expression.subExpression.name
+        for(uint i = 0; i < 42; i++) {
+        }
+    }
+}
+// ----
+// TestCase1: <
+// TestCase2: 1
+// TestCase3: i
```

Comment:
I see a little problem with verbosity here. The way it is now you have to define a new ID for every property you want to test. So when writing such a test, I'll probably end up with something like this:
```solidity
/// MyLoopInitialValueProperty1: initializationExpression.initialValue.property1
/// MyLoopInitialValueProperty2: initializationExpression.initialValue.property2
/// MyLoopInitialValueProperty3: initializationExpression.initialValue.property3
for (...) { ... }

...
// ----
// MyLoopInitialValueProperty1: x
// MyLoopInitialValueProperty2: y
// MyLoopInitialValueProperty3: z
```

which is very verbose and will get annoying quickly.

IMO a much better way to do this would be:
```solidity
/// MyLoop
for (...) { ... }
...
// ----
// MyLoop.initialValue.property1: x
// MyLoop.initialValue.property2: y
// MyLoop.initialValue.property3: z
```

What do you think @ekpyron? I know you wanted to get this merged quickly not to delay the PR it depends on, but maybe we could at least consider it as a refactor once this gets merged?

Reply:
For what it's worth, I think the current approach is more expressive, and more readable when looking at the test. On the other hand, it's a bit more annoying to write, but I think it's a fair price to pay.

Reply:
Hmm... why is it more expressive? It seems equivalent to me. I can't think of a case you would be able to do with one but not the other.

I also, don't think the more verbose version is more readable. It just has a lot of repetition that does not add anything of value. It requires you to invent more names, and these will either be short and hide detail or long and too verbose.

Reply:
Just as a note on this: the current design is fine and we should merge it. We can still refactor it afterwards.

Reply:
Another point to consider about design, but which could be done in the future, is regarding properties which are inside or are themselves an array of nodes.
For example, the name of the return variable of a function is inside property `returnParameters.parameters` which is a list. Another example, the name of parameters of a function is also inside a property which is a list.

Reply:
An easy way to handle that would be to allow expectation values to be arbitrary JSON. For easy parsing we would require them to be on a single line (i.e. no pretty-printing), which would also discourage trying to stuff things that are too complex in there. Would suffice for simple arrays at least. It's outside of the scope of this PR though.